include/drivers/pwm.h

```diff
@@ -89,6 +177,138 @@ static inline int z_impl_pwm_pin_set_cycles(const struct device *dev,
 	return api->pin_set(dev, pwm, period, pulse, flags);
 }
 
+#ifdef CONFIG_PWM_CAPTURE_API
+/**
+ * @brief Configure PWM period/pulse width capture for a single PWM input.
+ *
+ * After configuring PWM capture using this function, the capture can be
+ * enabled/disabled using @a pwm_pin_enable_capture() and @a
+ * pwm_pin_disable_capture().
+ *
+ * Note: This API function cannot be invoked from user space due to the use of a
+ * function callback. In user space, one of the simpler API functions (@a
+ * pwm_pin_capture_cycles(), @a pwm_pin_capture_usec(), or @a
+ * pwm_pin_capture_nsec()) can be used instead.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pwm PWM pin.
+ * @param flags PWM capture flags
+ * @param cb Application callback handler function to be called upon capture
+ * @param user_data User data to pass to the application callback handler
+ *                  function
+ *
+ * @retval -EINVAL if invalid function parameters were given
+ * @retval -ENOTSUP if PWM capture is not supported or the given flags are not
+ *                  supported
+ * @retval -EIO if IO error occurred while configuring
+ * @retval -EBUSY if PWM capture is already in progress
+ */
+static inline int pwm_pin_configure_capture(const struct device *dev,
+					    uint32_t pwm,
+					    pwm_flags_t flags,
+					    pwm_capture_callback_handler_t cb,
+					    void *user_data)
+{
+	struct pwm_driver_api *api;
```

Comment:
Why not combine the initialization into the declaration?

Reply:
Adopted.